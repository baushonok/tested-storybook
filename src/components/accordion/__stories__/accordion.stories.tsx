import { storiesOf } from "@storybook/react";
import React from "react";

import Accordion from "../index";
import { data } from "../__mocks__/data";

storiesOf("Accordion", module)
  .add("Default", () => (
    <div className="story-container">
      <Accordion>{data}</Accordion>
    </div>
  ))
  .add("No Content", () => (
    <div className="story-container">
      <Accordion></Accordion>
    </div>
  ));
