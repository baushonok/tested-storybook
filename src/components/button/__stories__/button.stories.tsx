import { storiesOf } from '@storybook/react';
import React from 'react';

import Button from '../button';

import '../../../common/styles.css';

storiesOf('Button', module).add('Default', () => <Button>Click me!</Button>);
