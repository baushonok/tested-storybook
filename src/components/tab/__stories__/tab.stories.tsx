import { storiesOf } from "@storybook/react";
import React from "react";

import Tab from "../index";

storiesOf("Tab", module)
  .add("Default", () => (
    <div className="story-container">
      <Tab name="First" content={<h1>content title</h1>} />
    </div>
  ))
  .add("Initially open", () => (
    <div className="story-container">
      <Tab name="Tab name" content={<h1>content title</h1>} isOpen={true} />
    </div>
  ))
  .add("Several tabs", () => (
    <div className="story-container">
      <Tab name="First" content={<h1>first content title</h1>} isOpen={true} />
      <Tab name="Second" content={<h1>second content title</h1>} />
      <Tab name="Third" content={<h1>third content title</h1>} />
    </div>
  ));
