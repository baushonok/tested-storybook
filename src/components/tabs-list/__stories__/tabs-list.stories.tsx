import { storiesOf } from '@storybook/react';
import React from 'react';

import TabsList from '../index';
import { data } from '../__mocks__/data';

storiesOf('TabsList', module).add('Default', () => (
  <div className="story-container">
    <TabsList data={data} />
  </div>
));
