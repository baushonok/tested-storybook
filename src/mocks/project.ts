import { ITab } from "../components/tabs-list/types";

export const data: ITab[] = [
  {
    name: "Common settings",
    content: "Common settings content",
  },
  {
    name: "Other settings",
    content: "Other settings content",
  },
];
