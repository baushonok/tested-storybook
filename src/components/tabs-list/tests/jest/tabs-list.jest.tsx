import React from "react";
import renderer from "react-test-renderer";

import TabsList from "../..";
import { data } from "../../__stories__/mocks";

test("TabsList should exist", () => {
  const component = renderer.create(<TabsList data={data} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
