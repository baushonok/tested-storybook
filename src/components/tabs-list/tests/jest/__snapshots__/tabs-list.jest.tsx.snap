// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsList shapshots should match when empty 1`] = `
<section
  className="tabs-list"
>
  <ul
    className="tabs-list__headers"
    data-testid="tab-toggle-list"
  />
  <ul
    className="tabs-list__content"
    data-testid="tab-content-list"
  />
</section>
`;

exports[`TabsList shapshots should match with data 1`] = `
<section
  className="tabs-list"
>
  <ul
    className="tabs-list__headers"
    data-testid="tab-toggle-list"
  >
    <button
      className="tab-toggle tab-toggle_active"
      onClick={[Function]}
      type="button"
    >
      First tab
    </button>
    <button
      className="tab-toggle "
      onClick={[Function]}
      type="button"
    >
      Second tab
    </button>
    <button
      className="tab-toggle "
      onClick={[Function]}
      type="button"
    >
      Third tab
    </button>
  </ul>
  <ul
    className="tabs-list__content"
    data-testid="tab-content-list"
  >
    <section
      className="tab-content "
    >
      First tab content
    </section>
    <section
      className="tab-content tab-content_hidden"
    >
      Second tab content
    </section>
    <section
      className="tab-content tab-content_hidden"
    >
      Third tab content
    </section>
  </ul>
</section>
`;
